BDArmory/ModuleTurret.cs

```diff
@@ -173,27 +175,24 @@ public void AimToTarget(Vector3 targetPosition, bool pitch = true, bool yaw = tr
 
-            Vector3 localTargetYaw =
-                yawTransform.parent.InverseTransformPoint(targetPosition - (yawTargetOffset*pitchTransform.right));
-            Vector3 targetYaw = Vector3.ProjectOnPlane(localTargetYaw, Vector3.up);
-            float targetYawAngle = VectorUtils.SignedAngle(Vector3.forward, targetYaw, Vector3.right);
-            targetYawAngle = Mathf.Clamp(targetYawAngle, -yawRange/2, yawRange/2);
-
-            Quaternion currYawRot = yawTransform.localRotation;
-            yawTransform.localRotation = Quaternion.Euler(0, targetYawAngle, 0);
-            Vector3 localTargetPitch =
-                pitchTransform.parent.InverseTransformPoint(targetPosition - (pitchTargetOffset*pitchTransform.up));
-            yawTransform.localRotation = currYawRot;
-            localTargetPitch.z = Mathf.Abs(localTargetPitch.z); //prevents from aiming wonky if target is behind
-            Vector3 targetPitch = Vector3.ProjectOnPlane(localTargetPitch, Vector3.right);
-            float targetPitchAngle = VectorUtils.SignedAngle(Vector3.forward, targetPitch, Vector3.up);
-            targetPitchAngle = Mathf.Clamp(targetPitchAngle, minPitch, maxPitch);
-
-            float yawOffset = Vector3.Angle(yawTransform.parent.InverseTransformDirection(yawTransform.forward),
-                targetYaw);
-            float currentYawSign = Mathf.Sign(Vector3.Dot(yawTransform.localRotation*Vector3.forward, Vector3.right));
-            float pitchOffset = Vector3.Angle(pitchTransform.parent.InverseTransformDirection(pitchTransform.forward),
-                targetPitch);
+            Vector3 yawNormal = yawTransform.up;
+            Vector3 yawComponent = Vector3.ProjectOnPlane(targetDirection, yawNormal);
+            Vector3 pitchNormal = Vector3.Cross(yawComponent, yawNormal);
+            Vector3 pitchComponent = Vector3.ProjectOnPlane(targetDirection, pitchNormal);
+
+            float currentYaw = yawTransform.localEulerAngles.y;
+            float yawError = VectorUtils.SignedAngleDP(
+                Vector3.ProjectOnPlane(referenceTransform.forward, yawNormal), 
+                yawComponent, 
+                Vector3.Cross(yawNormal, referenceTransform.forward));
+            float yawOffset = Mathf.Abs(yawError);
+            float targetYawAngle = Mathf.Clamp((currentYaw + yawError + 180f) % 360f - 180f, -yawRange / 2, yawRange / 2); // clamped target yaw
+
+            float currentPitch = -((pitchTransform.localEulerAngles.x + 180f) % 360f - 180f); // from current rotation transform
+            float realPitch = 90f - (float)Vector3d.Angle(yawNormal, referenceTransform.forward);
+            float targetPitchAngle = 90f - (float)Vector3d.Angle(yawNormal, pitchComponent) + (realPitch - currentPitch); // simple angle from yaw normal
+            float pitchOffset = Mathf.Abs(targetPitchAngle - currentPitch);
+            targetPitchAngle = Mathf.Clamp(targetPitchAngle, minPitch, maxPitch); // clamp pitch
```

Comment:
Aiming is now based on the orientation of the fire transform instead of the orientation of the rotation gizmos. Because with the old way some turrets aimed sideways. :( I'm looking at you BL9_14.